### DISCUSSION TITLE
Testing v-model type matching error/warning when using TypeScript

### DISCUSSION THREAD

**Discussion by LondonAppDev** [vuejs/test-utils#1969] [Q&A] [answered]
I'm testing that a component correctly updates a `v-model` value after clicking a button. 

I followed [the documentation for testing `v-model`](https://test-utils.vuejs.org/guide/advanced/v-model.html) and I've used the suggested approach of adding an `onUpdate:modelValue` prop like this:

```ts
const wrapper = mount(Editor, {
    props: {
      modelValue: 'initialText',
      'onUpdate:modelValue': (e) => wrapper.setProps({ modelValue: e })
    }
  })
```

The issue is that I get a type warning, because my components (eg: `Editor`) does not specify `onUpdate:modelValue` as a prop in `defineProps`.

What would be the correct way to handle this type warning properly? Is there any other approach except to use `(Editor as any)`?

**Comment by cexbrayat** [comment, marked as answer]
Hi @LondonAppDev 

We usually add `emits: ['update:modelValue'],` to the `Editor` component, and I think it should compile properly.
If that works as I expect, I'll add a note in the docs 👍

**Reply by LondonAppDev** [reply]
You're right, it does work. Thanks for the reply and apologies for taking your time.

**Reply by cexbrayat** [reply]
No worries. I updated the docs as it is not easy to figure out.
